packages/react-resize-handle/src/hooks/useMouseHandler.ts

```diff
@@ -98,7 +122,18 @@ export function useMouseHandler(params: UseMouseHandlerParams) {
       targetDocument?.addEventListener('touchmove', onDrag);
     }
 
-    params.onDragStart?.(event);
+    params.onDragStart?.(
+      event,
+      'touches' in event
+        ? {
+            type: EVENTS.touch,
+            event: event as TouchEvent,
+          }
+        : {
+            type: EVENTS.mouse,
+            event: event as MouseEvent,
+          }
+    );
```

Comment:
nit: probably we can add a specific local helper for that condition ('touches' in event)